src/units/abilities.cpp

```diff
@@ -1107,6 +1107,21 @@ namespace { // Helpers for attack_type::special_active()
 //beneficiary unit does not have a corresponding weapon
 //(defense against ranged weapons abilities for a unit that only has melee attacks)
 
+static bool overwrite_special_affects(const config& special)
+{
+	//log_scope("overwrite_special_affects_self");
+	const std::string& apply_to = special["overwrite_specials"];
+	if ( apply_to.empty() )
+		return false;
+	if ( apply_to == "none" )
+		return false;
+	if ( apply_to == "one_side" )
+		return true;
+	if ( apply_to == "both_sides" )
+		return true;
+	return false;
```

Comment:
This method can be simplified to `return apply_to == "one_side" || apply_to == "both_sides"`.